perl/lib/YAMLScript.pm

```diff
@@ -4,45 +4,78 @@ use strict;
 use warnings;
 
 our $VERSION = '0.1.21';
-our $libyamlscript_version = '0.1.34';
 
 use Alien::YAMLScript;
-use FFI::Platypus;
 use FFI::CheckLib ();
+use FFI::Platypus;
 use JSON ();
 
-sub new {
-    bless {}, shift;
-}
-
-unless ( Alien::YAMLScript->exact_version($libyamlscript_version) ) {
-    my $have = Alien::YAMLScript->version;
-    die "YAMLScript $VERSION requires Alien::YAMLScript libyamlscript version\n" .
-        "$libyamlscript_version, but you have $have";
-}
+# Alien::YAMLScript finds the proper libyamlscript version, but we need to be
+# using the proper version of Alien::YAMLScript:
+die "\$YAMLScript::VERSION ($YAMLScript::VERSION) and " .
+    "\$Alien::YAMLScript::VERSION($Alien::YAMLScript::VERSION) " .
+    "must be the same version"
+    unless $YAMLScript::VERSION eq $Alien::YAMLScript::VERSION;
 
+# Set up FFI functions:
 my $ffi = FFI::Platypus->new(
     api => 2,
     lib => [ Alien::YAMLScript->dynamic_libs ],
 );
 
-$ffi->function( graal_create_isolate => [qw( opaque opaque* opaque* )] => 'int' )
-    ->( undef, \my $isolate, \my $thread)
-    and die 'Could not setup evaluation';
+my $graal_create_isolate = $ffi->function(
+    graal_create_isolate =>
+        ['opaque', 'opaque*', 'opaque*'] => 'int',
+);
 
-$ffi->attach(
-    [ load_ys_to_json => 'load' ] => [qw( sint64 string )] => 'string' => sub {
-        my ( $xsub, $self, $ys ) = @_;
-        my $resp = JSON::decode_json( $xsub->( $thread, $ys ) );
+my $graal_tear_down_isolate = $ffi->function(
+    graal_tear_down_isolate =>
+        ['opaque*'] => 'int',
```

Comment:
We are not passing a pointer to a pointer, just a pointer
```suggestion
        ['opaque'] => 'int',
```
That fixes the problem with the error during destruction when the test runs

Reply:
`\o/`